src/space/objects/realization/blackhole.js

```diff
@@ -0,0 +1,91 @@
+import Vector from "../../utils/vector";
+import EncounterObject from "../pattern/encounterObject";
+import Type from "../pattern/encounterObjectType";
+import MyMath from "../../utils/myMath";
+import AssetLoader from "../../utils/assetLoader";
+
+const BH_SPRITE_RADIUS = 50;
+const BH_FORCE_SCALE = 600.000 / 1000.000;
+const BH_GRAVITY_RANGE = 8000;
+const BH_WRAP_INTERVAL = 100;
+
+export default class BlackHole extends EncounterObject{
+
+    /**
+      * Construction for black hole game element
+      *
+      * @param {Encounter} encounter
+      * @param {Vector} position - black hole's position
+      * @reference Black Hole (Warp) by Timothy Courtney (timothycourtney.io)
+      *  Licensed under Creative Commons: By Attribution 3.0 License
+      *  http://creativecommons.org/licenses/by/3.0/
+      */
+    constructor(encounter, position){
+        super(encounter, Type.BLACK_HOLE, BH_GRAVITY_RANGE, position);
+        this.bhImage = AssetLoader.getAsset("black-hole");
+        this.coordinateXInImage = 0;
+        this.coordinateYInImage = 0;
+        this.wrapInterval = BH_WRAP_INTERVAL;
+    }
+
+    /**
+      * generate a random position blackhole
+      *
+      * @param {Object} encounter
+      * @return {BlackHole} a new black hole with random position
+      */
+    static generateAtRandomPosition(encounter){
+        let x = Math.floor(Math.random()*encounter.width);
+        let y = Math.floor(Math.random()*encounter.height);
+        return new BlackHole(encounter, new Vector(x,y));
+    }
+
+    /**
+      * @param {Object} the object waiting for check
+      * @return {Boolean} whether other object collided with black hole
+      */
+    areColliding(others){
```

Comment:
I think it might get lost in the Discord channel, but `EncounterObject`  no longer has this method. It has been refactored to `collidesWith(other)`. Please, rewrite it so correct method is referenced.